File: app/src/main/java/rectangledbmi/com/pittsburghrealtimetracker/patterns/stops/rendering/StopRequestAccumulator.java
package rectangledbmi.com.pittsburghrealtimetracker.patterns.stops.rendering;

import java.util.List;

import rectangledbmi.com.pittsburghrealtimetracker.patterns.stops.either.FullStopSelectionState;
import rectangledbmi.com.pittsburghrealtimetracker.patterns.stops.either.MapState;

/**
 * <p>DTO for a pattern view model
 *    to store the selection state, map state, and immediate changes for stop visibility.</p>
 * <p>Created by epicstar on 10/8/16.</p>
 * @author Jeremy Jao
 * @since 78
 */

public class StopRequestAccumulator {
    private final FullStopSelectionState fullStopSelectionState;
    private final MapState mapState;
    private final List<StopRenderRequest> stopsToChange;

    private StopRequestAccumulator(FullStopSelectionState fullStopSelectionState, MapState mapState, List<StopRenderRequest> stopsToChange) {
        this.fullStopSelectionState = fullStopSelectionState;
        this.mapState = mapState;
        this.stopsToChange = stopsToChange;
    }

    public static StopRequestAccumulator create(
            FullStopSelectionState fullStopSelectionState,
            MapState mapState,
            List<StopRenderRequest> stopsToChange) {
        return new StopRequestAccumulator(fullStopSelectionState, mapState, stopsToChange);
    }

    /**
     * @return The most recent selection state for stops
     */
    public FullStopSelectionState getFullStopSelectionState() {
        return fullStopSelectionState;
    }

    /**
     * @return The most recent map state
     */
    public MapState getMapState() {
        return mapState;
    }

    /**
     * @return the processed list of stops to change
     */
    public List<StopRenderRequest> getStopsToChange() {
        return stopsToChange;
    }
}

File: app/src/main/java/rectangledbmi/com/pittsburghrealtimetracker/patterns/stops/rendering/StopRequestAccumulator.kt
package rectangledbmi.com.pittsburghrealtimetracker.patterns.stops.rendering

import rectangledbmi.com.pittsburghrealtimetracker.patterns.stops.either.FullStopSelectionState
import rectangledbmi.com.pittsburghrealtimetracker.patterns.stops.either.MapState

/**
 *
 * DTO for a pattern view model
 * to store the selection state, map state, and immediate changes for stop visibility.
 *
 * Created by epicstar on 10/8/16.
 * @author Jeremy Jao
 * @since 78
 */

class StopRequestAccumulator private constructor(
        /**
         * @return The most recent selection state for stops
         */
        val fullStopSelectionState: FullStopSelectionState?,
        /**
         * @return The most recent map state
         */
        val mapState: MapState?,
        /**
         * @return the processed list of stops to change
         */
        val stopsToChange: List<StopRenderRequest>) {
    companion object {

        fun create(
                fullStopSelectionState: FullStopSelectionState?,
                mapState: MapState?,
                stopsToChange: List<StopRenderRequest>): StopRequestAccumulator {
            return StopRequestAccumulator(fullStopSelectionState, mapState, stopsToChange)
        }
    }
}
